### PR TITLE
[FIXED JENKINS-29221] WorkflowRun assigned FlowExecution when execution start fails

### PR DESCRIPTION
**Pull request by tfennelly**
[See JENKINS-29221](https://issues.jenkins-ci.org/browse/JENKINS-29221).

@reviewbybees

### CODE REVIEW

File: job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java [hunk 1]
```diff
@@ -180,13 +180,18 @@ public WorkflowRun(WorkflowJob job, File dir) throws IOException {
...
             completed = new AtomicBoolean();
             logsToCopy = new LinkedHashMap<String,Long>();
-            execution.start();
-            executionPromise.set(execution);
+            newExecution.start();
+            
+            // It's only okay to initialise the global once the new FlowExecution
+            // has successfully started.
+            execution = newExecution;
+            executionPromise.set(newExecution);
```
Comment (tfennelly) [inline job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java]: @kohsuke in the situation where the `execution` fails to start, does the instance need to be cleaned up in some way?

**Comment by tfennelly** [comment]
@kohsuke KK, would you be able to review this please?

File: job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java (hunk 1 above)
Comment (kohsuke) [inline job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java]: No, no such cleanup needed.

**Comment by kohsuke** [comment]
:bee: this change is good from my PoV.

**Comment by tfennelly** [comment]
@reviewbybees done

**Comment by tfennelly** [comment]
@KostyaSha Yeah... I'm just following the pattern already established in that code (see other `@Test` methods). While I too (like you) don't like the "all on one line" style, I don't agree with breaking with what's already there and everyone doing what they think is "right".

**Comment by KostyaSha** [comment]
@tfennelly seems one line annotation is only @jglick coding style. Just want to see some consistency... :(

**Comment by tfennelly** [comment]
@KostyaSha right, I'm not sure of anyone else doing it. I don't love it, but it doesn't bother me that much either. I've seen lots of different code style conventions used in Jenkins code (core and in plugins), so I've adopted the policy of following/respecting whatever the "local" convention is as best I can.

**Comment by jglick** [comment]
FWIW my convention is to put only determiner annotations like `@Test` (where in English you could put the word “the” in front of the annotation: “the test `worksOnMyMachine`”—there can be only one such) on the same line, but to put descriptive annotations like `@Issue` on their own lines.

**Comment by jglick** [comment]
Do we not need to call some variant of `finish` if creating the execution fails? To fire build finish events, etc. That is what I expected the fix of this issue to involve.

**Pull request merged**